.github/workflows/ci.yml

```diff
@@ -4,6 +4,25 @@ on:
    branches: [master, develop]
 
 jobs:
+  treefmt:
+   name: Run treefmt
+   environment: cachix # for secrets
+   runs-on: ubuntu-latest
+   steps:
+     - uses: actions/checkout@v2
+       with:
+         submodules: true
+     - uses: cachix/install-nix-action@v14.1
+     - uses: cachix/cachix-action@v10
+       with:
+         name: wire-server
+         signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+     - name: Install treefmt and nixpkgs-fmt (without pulling in all of dev-env)
+       run: nix-env -if nix/default.nix -iA pkgs.treefmt pkgs.nixpkgs-fmt pkgs.shellcheck
+     - name: Run treefmt
+       run: treefmt
```

Comment:
This github action is unnecessary, as linters already run `make lint-all` here: https://github.com/zinfra/cailleach/blob/ab3a921c7a6cc903c0d1d578040e05a566c84bfa/wire-server-private/ci/pipelines/jobs/linters.dhall#L40-L60

please remove this github actions block. Also, we agreed to not introduce more github actions if not strictly necessary.

Reply:
I'd like to keep these 20 lines of yaml.

 - They're not a big maintenance burden, as it simply invokes `treefmt` after installing the dependencies.
 - Very quick feedback (40 seconds)
 - No requirement to log into concourse, so useful feedback for external contributors


It would be a different thing if we'd start to model complicated logic in GH actions, but this simply executes a script to give semi-instant feedback.

It looks like Concourse also does it, because it's invoked in the `lint-all` target from the Makefile, but it's way less accessible.

Reply:
it spams emails and re-does something that is already done, giving twice the amount of linting feedback for the same issue. To me, this is just noise and unnecessary. Maybe someone else has an opinion here also.

Reply:
You can change if GH actions should send you emails on failure in [your notification settings](https://github.com/settings/notifications):

![image](https://user-images.githubusercontent.com/183879/200570368-4396b934-a6e9-4433-b1b2-df6976ca749a.png)